### PR TITLE
Null safety migration of packages/flutter_tools/test/commands.shard/hermetic, part 2/3

### PR DESCRIPTION
**Pull request by liamappelbe**
Null safety migration of packages/flutter_tools/test/commands.shard/hermetic, part 2/3

### CODE REVIEW

**Comment by christopherfujino** [comment]
This has test failures

**Comment by liamappelbe** [comment]
Tests are all passing now

File: packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart [hunk 1]
```diff
@@ -124,7 +122,7 @@ void main() {
 
       expect(response.data['id'], 0);
       expect(response.data['result'], isNotEmpty);
-      expect((response.data['result'] as Map<String, dynamic>)['platforms'], <String>{'macos'});
+      expect((response.data['result'] as Map<String, dynamic>?)!['platforms'], <String>{'macos'});
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart:125]:
```suggestion
      expect((response.data['result']! as Map<String, dynamic>)['platforms'], <String>{'macos'});
```

File: packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart [hunk 2]
```diff
@@ -57,7 +55,7 @@ class FakeDaemonStreams implements DaemonStreams {
   }
 
   @override
-  void send(Map<String, dynamic> message, [ List<int> binary ]) {
+  void send(Map<String, dynamic> message, [ List<int>? binary ]) {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart:58]:
```suggestion
  void send(Map<String, Object?> message, [ List<int>? binary ]) {
```

Here and throughout this file, can you update `dynamic` -> `Object?` (or `Object` where possible) per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var-and-dynamic

File: packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart [hunk 3]
```diff
@@ -137,11 +135,11 @@ void main() {
       );
       globals.printError('daemon.logMessage test');
       final DaemonMessage response = await daemonStreams.outputs.stream.firstWhere((DaemonMessage message) {
-        return message.data['event'] == 'daemon.logMessage' && (message.data['params'] as Map<String, dynamic>)['level'] == 'error';
+        return message.data['event'] == 'daemon.logMessage' && (message.data['params'] as Map<String, dynamic>?)!['level'] == 'error';
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart:138]:
```suggestion
        return message.data['event'] == 'daemon.logMessage' && (message.data['params']! as Map<String, Object?>)['level'] == 'error';
```

File: packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart [hunk 4]
```diff
@@ -155,11 +153,11 @@ void main() {
       );
       globals.printWarning('daemon.logMessage test');
       final DaemonMessage response = await daemonStreams.outputs.stream.firstWhere((DaemonMessage message) {
-        return message.data['event'] == 'daemon.logMessage' && (message.data['params'] as Map<String, dynamic>)['level'] == 'warning';
+        return message.data['event'] == 'daemon.logMessage' && (message.data['params'] as Map<String, dynamic>?)!['level'] == 'warning';
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart:156]:
```suggestion
        return message.data['event'] == 'daemon.logMessage' && (message.data['params']! as Map<String, Object?>)['level'] == 'warning';
```

File: packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart [hunk 5]
```diff
@@ -448,7 +446,7 @@ void main() {
         final DaemonMessage startAppResponse = await broadcastOutput.firstWhere(_notEvent);
         expect(startAppResponse.data['id'], 1);
         expect(device.startAppPackage, applicationPackage);
-        final Map<String, dynamic> startAppResult = startAppResponse.data['result'] as Map<String, dynamic>;
+        final Map<String, dynamic> startAppResult = (startAppResponse.data['result'] as Map<String, dynamic>?)!;
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart:449]:
```suggestion
        final Map<String, Object?> startAppResult = startAppResponse.data['result']! as Map<String, Object?>;
```

File: packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart [hunk 6]
```diff
@@ -519,7 +517,7 @@ void main() {
       unawaited(daemonStreams.outputs.stream
         .firstWhere((DaemonMessage request) => request.data['method'] == 'app.exposeUrl')
         .then((DaemonMessage request) {
-          expect((request.data['params'] as Map<String, dynamic>)['url'], equals(originalUrl));
+          expect((request.data['params'] as Map<String, dynamic>?)!['url'], equals(originalUrl));
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart:520]:
```suggestion
          expect((request.data['params']! as Map<String, Object?>)['url'], equals(originalUrl));
```

File: packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart [hunk 7]
```diff
@@ -536,7 +534,7 @@ void main() {
 
       daemonStreams.inputs.add(DaemonMessage(<String, dynamic>{'id': 0, 'method': 'devtools.serve'}));
       final DaemonMessage response = await daemonStreams.outputs.stream.firstWhere((DaemonMessage response) => response.data['id'] == 0);
-      final Map<String, dynamic> result = response.data['result'] as Map<String, dynamic>;
+      final Map<String, dynamic> result = (response.data['result'] as Map<String, dynamic>?)!;
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart:537]:
```suggestion
      final Map<String, Object?> result = response.data['result']! as Map<String, Object?>;
```

File: packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart [hunk 8]
```diff
@@ -552,7 +550,7 @@ void main() {
...
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/daemon_test.dart:553]:
```suggestion
      final Map<String, Object?> result = response.data['result']! as Map<String, Object?>;
```

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**